R/navbar.R

```diff
@@ -11,13 +11,11 @@ data_navbar <- function(pkg = ".", depth = 0L) {
   components[names(components_meta)] <- components_meta
   components <- purrr::compact(components)
 
-  # Any unplaced components go to the right of the left navbar
   right_comp <- intersect(structure$right, names(components))
   left_comp <- intersect(structure$left, names(components))
-  extra_comp <- setdiff(names(components), c(left_comp, right_comp))
 
   # Backward compatiblity
-  left <- navbar$left %||% components[c(left_comp, extra_comp)]
+  left <- navbar$left %||% components[left_comp]
   right <- navbar$right %||% components[right_comp]
 
   list(
```

Comment:
If we ever have to make this code more complicated in the future, we should move the `render_navbar_links()` call later, so we can test the structure independent of the HTML it generates. We don't need to do this today, but if the tests get more complex it would be worth it.